x-pack/plugins/infra/public/alerting/log_threshold/components/alert_details_app_section/index.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ALERT_DURATION, ALERT_END } from '@kbn/rule-data-utils';
+import moment from 'moment';
+import React from 'react';
+import { type PartialCriterion } from '../../../../../common/alerting/logs/log_threshold';
+import { CriterionPreview } from '../expression_editor/criterion_preview_chart';
+import { AlertDetailsAppSectionProps } from './types';
+
+const AlertDetailsAppSection = ({ rule, alert }: AlertDetailsAppSectionProps) => {
+  const ruleWindowSizeMS = moment
+    .duration(rule.params.timeSize, rule.params.timeUnit)
+    .asMilliseconds();
+  const alertDurationMS = alert.fields[ALERT_DURATION]! / 1000;
+  const TWENTY_TIMES_RULE_WINDOW_MS = 20 * ruleWindowSizeMS;
+  /**
+   * This is part or the requirements (RFC).
+   * If the alert is less than 20 units of `FOR THE LAST <x> <units>` then we should draw a time range of 20 units.
+   * IE. The user set "FOR THE LAST 5 minutes" at a minimum we should show 100 minutes.
+   */
+  const rangeFrom =
+    alertDurationMS < TWENTY_TIMES_RULE_WINDOW_MS
+      ? Number(moment(alert.start).subtract(TWENTY_TIMES_RULE_WINDOW_MS, 'millisecond').format('x'))
+      : Number(moment(alert.start).subtract(ruleWindowSizeMS, 'millisecond').format('x'));
+
+  const rangeTo = alert.active
+    ? Date.now()
+    : Number(moment(alert.fields[ALERT_END]).add(ruleWindowSizeMS, 'millisecond').format('x'));
+
+  return (
+    // Create a chart per-criteria
+    <>
```

Comment:
**question:** is this fragment here in preparation for additional items to be added? Otherwise, you can directly return the mapping result with no need to invoke the fragment:
```js
return rule.params.criteria.map((criteria) => (
  <CriterionPreview
    ruleParams={rule.params}
    sourceId={rule.params.logView.logViewId}
    chartCriterion={criteria as PartialCriterion}
    showThreshold={true}
    executionTimeRange={{ gte: rangeFrom, lte: rangeTo }}
  />
))
```

Reply:
done